test/integration/Test/Integration/Framework/Request.hs

```diff
@@ -129,31 +133,34 @@ request'
         )
     => (Method, Text)
         -- ^ HTTP method and request path
+    -> Maybe RequestHeaders
+        -- ^ Request headers
```

Comment:
I'd suggest to use a `Map HeaderName Header` or `[RequestHeaders]` here instead and perform a union with the base / default headers later: `requestHeaders = baseHeaders <> extraHeaders`. This will allow overwriting the base headers when necessary and/or provide extra ones still fairly easily, or none using `mempty`. 

What do you think @piotr-iohk ?

Reply:
Not sure if I follow. `RequestHeaders` is just `[Header]` (and `Header` is `(HeaderName, ByteString)`) so why use `[ResuestHeaders]`?

Also how would it be possible to overwrite the `baseHeaders`? (I understand that providing `mempty` or `[]` (if I'll change to `[Header]`) would be having just base (as a result of `base <> mempty`), providing some `extraHeaders` would `add` base and extra but how to overwrite base then?)

Reply:
Using a `Map` and doing the union between two maps, we'll the values associated to keys present in the second map overriding the one corresponding to keys present in the first map (if present).

So:

```hs
>>> let m1 = Map.fromList [("a", 1), ("b", 2)]
>>> let m2 = Map.fromList [("a", 14), ("c", 42)]
>>> m1 <> m2
Map.fromList [("a", 14), ("b", 2), ("c", 42)]
```

Reply:
Hm. Well, in fact it is:
```haskell
Prelude> import qualified Data.Map as Map
Prelude Map> let m1 = Map.fromList [("a", 1), ("b", 2)]
Prelude Map> let m2 = Map.fromList [("a", 14), ("c", 42)]
Prelude Map> m1 <> m2
fromList [("a",1),("b",2),("c",42)]
```
, but that's ok as the arguments can be flipped :) :
... still it won't allow something like providing no headers at all...

Reply:
>  ... still it won't allow something like providing no headers at all...

True. Aaaaand, okay. You've convinced me ^^